src/download/exchangeDownloader.ts

```diff
@@ -206,39 +206,54 @@ export async function searchExchange(
 }
 
 /**
- * @description Looks at all versions of an api in exchange for an instance that matched the deployment regex
+ * @description Returns the version of an api in exchange from the instance fetched asset.version value
  *
  * @export
  * @param {string} accessToken
  * @param {RestApi} restApi
- * @param {RegExp} deployment
- * @returns {Promise<string>} Returned the version string that matches the regex passed.  Will return first found result
+ * @param {RegExp} [deployment]
+ * @returns {Promise<string>} Returned the version string from the instance fetched asset.version value
  */
 export async function getVersionByDeployment(
   accessToken: string,
   restApi: RestApi,
-  deployment: RegExp
+  deployment?: RegExp
 ): Promise<void | string> {
-  const asset = await getAsset(
-    accessToken,
-    `${restApi.groupId}/${restApi.assetId}`
-  );
+  if (deployment) {
+    ramlToolLogger.warn(
+      "Warning: The 'deployment' argument is deprecated and will be ignored due to changes in the RAML spec. This argument will be removed in the next major version."
```

Comment:
warning is not updated/modified?

Reply:
Good catch. I updated the message there as well.